Clarify transform for ViewStyleProps, include children in union prop docs

brentvatne opened the pull request:
Show the proper shape of `transform` props:

![screen shot 2015-05-06 at 6 05 12 pm](https://cloud.githubusercontent.com/assets/90494/7506832/7c1c2b4c-f41a-11e4-9ce7-971cd4d3b020.png)

Currently it is just `transform: [object]`, so this should make it much clearer in the docs as well as properly validate.

cc @vjeux @ide

ide commented:
Nice work there.

frantic commented on Libraries/Components/View/ViewStylePropTypes.js:
```diff
@@ -34,7 +34,15 @@ var ViewStylePropTypes = {
   shadowRadius: ReactPropTypes.number,
-  transform: ReactPropTypes.arrayOf(ReactPropTypes.object),
+  transform: ReactPropTypes.arrayOf(
+    ReactPropTypes.oneOfType([
```
Will this throw a warning if I use

```javascript
<View style={{transform: {rotation: 1, scaleX: 2}}} />
```
? (i.e. mixture of these shapes)

brentvatne replied:
@frantic - nope, but that won't work - you need to put each transform in a separate object. Also, I have realized that `rotation` is not a prop on iOS, it looks like that was just for Android. On iOS, the `rotate` transform property exists and accepts a string: `{rotate: "1deg"}` or `{rotate: "1rad"}` just like the CSS property.

So this is how that would be written:

```javascript
<View style={{transform: [{rotate: "1deg"}, {scaleX: 2}]}} />
```

If you were to writer this, rotate would be applied by scaleX would not:

```javascript
<View style={{transform: [{rotate: "1deg", scaleX: 2}]}} />
```

brentvatne commented:
d177a45bdce124b635dbc31ab0949d2e9d139eb3 broke the transform docs, rebased and fixed so this is now in a new section:

![screen shot 2015-05-07 at 12 18 24 pm](https://cloud.githubusercontent.com/assets/90494/7523903/58a7cd9e-f4b3-11e4-8a0b-f7d6ca30abed.png)
![screen shot 2015-05-07 at 12 19 26 pm](https://cloud.githubusercontent.com/assets/90494/7523905/5b1eab7e-f4b3-11e4-8836-5fa0c7c690c6.png)
